pact-data-model/src/lib.rs

```diff
@@ -13,17 +13,18 @@
 use chrono::{DateTime, Utc};
 use rust_decimal::Decimal;
 use schemars::schema::{
-    ArrayValidation, InstanceType, NumberValidation, Schema, SchemaObject, StringValidation,
+    ArrayValidation, InstanceType, NumberValidation, ObjectValidation, Schema, SchemaObject,
+    StringValidation,
 };
-use schemars::JsonSchema;
+use schemars::{JsonSchema, Map};
 use serde::{Deserialize, Serialize};
 use uuid::Uuid;
 
 mod schema_gen;
 pub use schema_gen::generate_schema;
 
 #[derive(Debug, Serialize, Deserialize, Clone, JsonSchema, PartialEq)]
-#[serde(rename_all = "camelCase")]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
```

Comment:
```suggestion
#[serde(rename_all = "camelCase")]
```

Reply:
Do we want to remove `deny_unknown_fields` though? Without it, random fields can be added to the JSON and it will still pass validation against our schema. Do we want to allow this?

Reply:
Through other channel it became clear that I was making an assumption here that does not correspond to anything explicitly stated in the tech specs. Thanks, @zeitgeist! 
Updated accordingly